WordPress/src/main/java/org/wordpress/android/ui/blaze/blazecampaigns/campaigndetail/CampaignDetailMapper.kt

```diff
@@ -0,0 +1,40 @@
+package org.wordpress.android.ui.blaze.blazecampaigns.campaigndetail
+
+import org.wordpress.android.R
+import org.wordpress.android.ui.blaze.BlazeFeatureUtils
+import org.wordpress.android.ui.utils.UiString
+import javax.inject.Inject
+
+class CampaignDetailMapper @Inject constructor(
+    private val blazeFeatureUtils: BlazeFeatureUtils,
+) {
+    fun toPrepared(url: String, addressToLoad: String) = CampaignDetailUiState.Prepared(
+        model = CampaignDetailModel(
+            enableJavascript = true,
+            enableDomStorage = true,
+            userAgent = blazeFeatureUtils.getUserAgent(),
+            enableChromeClient = true,
+            url = url,
+            addressToLoad = addressToLoad
+        )
+    )
+
+    fun toNoNetworkError(buttonClick: () -> Unit) = CampaignDetailUiState.Error(
+        title = UiString.UiStringRes(R.string.campaign_detail_no_network_error_title),
+        description = UiString.UiStringRes(R.string.campaign_detail_error_description),
+        button = CampaignDetailUiState.Error.ErrorButton(
+            text = UiString.UiStringRes(R.string.campaign_detail_error_button_text),
+            click = buttonClick
+        )
+    )
+
+    fun toGenericError(buttonClick: () -> Unit) = CampaignDetailUiState.Error(
+        title = UiString.UiStringRes(R.string.campaign_detail_error_title),
+        description = UiString.UiStringRes(R.string.campaign_detail_error_description),
+        button = CampaignDetailUiState.Error.ErrorButton(
+            text = UiString.UiStringRes(R.string.campaign_detail_error_button_text),
+            click = buttonClick
+        )
+    )
+}
```

Comment:
Hey @zwarm 
I like using a mapper for the prepared status 👍🏼 .
❓But do you think we need the mapper for the error state?
I tried this on my PR as well https://github.com/wordpress-mobile/WordPress-Android/pull/18842/commits. Once I did this, I felt that in the scenario of an error, we couldn't effectively test the type of error(Generic/No network) on the UI state. Then I moved the error types as a subclass of the error in this [commit ](https://github.com/wordpress-mobile/WordPress-Android/pull/18842/commits/056bb36c280085f7eed4462d9cff8213f6bdc209)

The benefit of dropping the `toNoNetworkError` and `toGenericError` and using data class to represent `NoNetworkError` and `GenericError` is that we can assert which type of error is in the UI state. See this [line](https://github.com/wordpress-mobile/WordPress-Android/pull/18842/files#diff-36428c2690917ef37687ab8574ab294a88ad37acd33f80f379c6cbe8d6723682R126). I don't have a strong opinion on both of this approach.  

One other approach we can do is to use the mapper function itself and use an error type value enum. Then assert on the ui test that the error type is  what we want. 

Would love to hear your thoughts

Reply:
> Then I moved the error types as a subclass of the error

I'm on board with this idea, definitely much easier to validate if we have the correct error view showing. I also thought of the same `enum` solution, but felt the data class was the way to go in this instance. I kept the mapper functions `toNoNetworkError` and `toGenericError`. They are simple to mock for testing.

Nice refactor!